File: Top 50 Coding Interview Questions/37. Write a code to find consecutive largest subsequence/findconsecutivelargestsubsequence.py
def LongestConseqSubseq(arr, l):
    # List to store lengths of consecutive subsequences
    val = []
    c = 0  # Counter to track the length of the current subsequence

    # Iterate through each element in the array
    for i in range(l):
        n = 1  # Initialize the step for consecutive checking

        # Check if consecutive elements exist in the array
        while arr[i] + n in arr:
            c += 1  # Increment counter for each consecutive element
            n += 1  # Move to the next possible consecutive element

        # Add the length of the current sequence to the list
        val.append(c + 1)
        c = 0  # Reset counter for the next element

    # Return the maximum length from the list of subsequence lengths
    return max(val)

# Input array
array = [7, 8, 1, 5, 4, 3]

# Output the length of the longest consecutive subsequence
print("Longest Consecutive Subsequence:", LongestConseqSubseq(array, len(array)))

File: Top 50 Coding Interview Questions/37. Write a code to find consecutive largest subsequence/findconsecutivelargestsubsequence.cpp
#include <bits/stdc++.h>
using namespace std;

// Function to find the length of the longest contiguous subsequence
int findLongestConseqSubseq(int arr[], int n) {
    // Use an unordered set to store unique elements for O(1) lookup
    unordered_set<int> elements(arr, arr + n);
    int longestStreak = 0; // Variable to track the maximum length

    // Iterate over the unique elements
    for (int num : elements) {
        // Check if 'num' is the start of a sequence
        if (elements.find(num - 1) == elements.end()) {
            int currentNum = num;    // Start of the sequence
            int currentStreak = 1;  // Length of the current sequence

            // Count the consecutive elements in the sequence
            while (elements.find(currentNum + 1) != elements.end()) {
                currentNum++;
                currentStreak++;
            }

            // Update the longest streak if the current sequence is longer
            longestStreak = max(longestStreak, currentStreak);
        }
    }

    return longestStreak; // Return the length of the longest contiguous subsequence
}

// Driver code
int main() {
    int arr[] = {1, 3, 2, 2};
    int n = sizeof(arr) / sizeof(arr[0]);

    // Output the result
    cout << "Length of the Longest Contiguous Subsequence is "
         << findLongestConseqSubseq(arr, n) << endl;

    return 0;
}
